src/helpers.js

```diff
@@ -117,25 +119,88 @@ export function isTraversable(tileID, me) {
   }
 }
 
+/* eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["bigGrid"] }] */
+
+/*
+ * place all values from smallGrid into bigGrid. Align the upper left corner at x, y
+ */
+export function fillGridWithSubgrid(bigGrid, smallGrid, x, y) {
+  const smallGridWidth = smallGrid.length;
+  const smallGridHeight = smallGrid[0].length;
+
+  for (let i = 0; i < smallGridWidth; i++) {
+    for (let j = 0; j < smallGridHeight; j++) {
+      bigGrid[i + x][j + y] = smallGrid[i][j];
+    }
+  }
+}
+
+
+/* Returns a 2d cell array of traversible (1) and blocked (0) cells inside a tile.
+ *
+ * tileIsTraversable: if this tile is traversable
+ * cpt: number of cells per tile
+ * objRadius: radius of untraversable object in pixels
+ */
+export function traversableCellsInTile(tileIsTraversable, cpt, objRadius) {
+  const gridTile = [];
+  // Start with all traversable
+  let i;
+  let j;
+  for (i = 0; i < cpt; i++) {
+    gridTile[i] = new Array(cpt);
+    for (j = 0; j < cpt; j++) {
+      gridTile[i][j] = 1;
+    }
+  }
+
+  if (!tileIsTraversable) {
+    const midCell = (cpt - 1.0) / 2.0;
+    for (i = 0; i < cpt; i++) {
+      for (j = 0; j < cpt; j++) {
+        const xDiff = Math.max(Math.abs(i - midCell) - 0.5, 0);
+        const yDiff = Math.max(Math.abs(j - midCell) - 0.5, 0);
+        const cellDist = Math.sqrt((xDiff * xDiff) + (yDiff * yDiff));
+        const ppc = PIXELS_PER_TILE / cpt; // number of pixels per cell
+        const pixelDist = cellDist * ppc;
+        if (pixelDist <= objRadius) {
+          // This cell touches the object, is not traversable
+          gridTile[i][j] = 0;
+        }
+      }
+    }
+  }
+  return gridTile;
+}
+
 
 /*
- * Returns a 2D array of traversable (1) and blocked (0) tiles.
+ * Returns a 2D array of traversable (1) and blocked (0) cells. Size of return grid is
+ * map.length * cpt
  *
  * The 2D array is an array of the columns in the game. empty_tiles[0] is
  * the left-most column. Each column array is an array of the tiles in
  * that column, with 1s and 0s.  empty_tiles[0][0] is the upper-left corner
  * tile.
+ *
+ * cpt: number of cells per tile
+ * map: 2D array representing the Tagpro map
+ * me: reference to the TagproBot
  */
-export function getTraversableTiles(tagproMap, me) {
-  const xl = tagproMap.length;
-  const yl = tagproMap[0].length;
-  const emptyTiles = [];
-
-  for (let x = 0; x < xl; x++) {
-    emptyTiles[x] = new Array(yl);
+export function getTraversableCells(cpt, map, me) {
+  const xl = map.length;
+  const yl = map[0].length;
+  const emptyCells = [];
+  let x;
+  for (x = 0; x < xl * cpt; x++) {
+    emptyCells[x] = new Array(yl * cpt);
+  }
+  for (x = 0; x < xl; x++) {
     for (let y = 0; y < yl; y++) {
```

Comment:
https://stackoverflow.com/questions/18465211/javascript-loop-variable-scope

"

" JavaScript doesn't have block scope, just function scope. Since the initialization of i is within one function, that variable is accessible anywhere else in that same function."

God that's so weird. I guess that answers our question :(